Orange/widgets/unsupervised/owhierarchicalclustering.py

```diff
@@ -944,8 +944,8 @@ def _update(self):
 
         if distances is not None:
             # Convert to flat upper triangular distances
-            i, j = numpy.triu_indices(distances.X.shape[0], k=1)
-            distances = distances.X[i, j]
+            i, j = numpy.triu_indices(distances.shape[0], k=1)
+            distances = numpy.asarray(distances[i, j])
```

Comment:
It has a shape (N, N). And I foresaw the frequent need of upper triangular, so `DistMatrix.flat()` is also provided. `triu()` might be better, now that I think about it.